src/commands/generateMarkdownDocument.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { IActionContext, UserCancelledError } from "@microsoft/vscode-azext-utils";
+import * as fs from "fs";
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { ext } from "../extensionVariables";
+import { ApiVersionDefinitionTreeItem } from "../tree/ApiVersionDefinitionTreeItem";
+import { UiStrings } from "../uiStrings";
+import { getApiSpecification } from "../utils/apiSpecificationUtils";
+// @ts-ignore
+import * as widdershins from "widdershins";
+import { makrdownDocuments } from "../constants";
+import { writeToTemporaryFile } from "../utils/fsUtil";
+import { OpenApiUtils } from "../utils/openApiUtils";
+
+export async function generateMarkdownDocument(context: IActionContext, node?: ApiVersionDefinitionTreeItem) {
+    let apiSpecification;
+    if (node) {
+        apiSpecification = node;
+    } else {
+        apiSpecification = await getApiSpecification(UiStrings.SelectApiSpecification, context);
+        if (!apiSpecification) {
+            throw new UserCancelledError();
+        }
+    }
+
+    let apiSpecificationContent: string;
+    let folderName: string;
+    let fileName: string;
+    if (apiSpecification instanceof ApiVersionDefinitionTreeItem) {
+        apiSpecificationContent = await ext.openApiEditor.getData(apiSpecification);
+        folderName = `${apiSpecification.apiCenterName}-${apiSpecification.apiCenterApiName}`;
+        fileName = `${apiSpecification.apiCenterApiVersionName}-tempFile.md`;
+    } else {
+        apiSpecificationContent = await fs.promises.readFile(apiSpecification.fsPath, { encoding: 'utf-8' });
+        folderName = makrdownDocuments;
+        fileName = `${path.basename(apiSpecification.fsPath)}-tempFile.md`;
+    }
+
+    await showMarkdownDocument(apiSpecificationContent, folderName, fileName);
+}
+
+async function showMarkdownDocument(apiSpecificationContent: string, folderName: string, fileName: string) {
+    const markdownOutput = await convertApiToMarkdown(apiSpecificationContent);
+
+    const fileUri = await writeToTemporaryFile(markdownOutput, folderName, fileName);
+
+    await vscode.window.showTextDocument(fileUri);
```

Comment:
the excuteCommand in next line will now show fileUri?

Reply:
`vscode.window.showTextDocument(fileUri)` will show raw file.
`vscode.commands.executeCommand('markdown.showPreviewToSide', fileUri)` will preview markdown side by side.